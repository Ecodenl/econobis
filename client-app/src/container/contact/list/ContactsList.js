import React, { Component } from 'react';

import DataTable from '../../../components/dataTable/DataTable';
import DataTableHead from '../../../components/dataTable/DataTableHead';
import DataTableBody from '../../../components/dataTable/DataTableBody';
import ContactsListHead from './ContactsListHead';
import ContactsListFilter from './ContactsListFilter';
import ContactsListItem from './ContactsListItem';
import ContactsDeleteItem from './ContactsDeleteItem';

class ContactsList extends Component {
    constructor(props){
        super(props);

        this.state = {
            showDeleteItem: false,
            deleteItem: {
                id: '',
                fullName: ''
            }
        };
    }

    // On key Enter filter form will submit
    handleKeyUp = (e) => {
        if (e.keyCode === 13) {
            this.props.onSubmitFilter();
        }
    };

    showDeleteItemModal = (id, name) => {
        this.setState({
            ...this.state,
            showDeleteItem: true,
            deleteItem:{
                ...this.state.deleteItem,
                id: id,
                fullName: name
            }
        });
    };

    closeDeleteItemModal = () => {
        this.setState({
            ...this.state,
            showDeleteItem: false,
            deleteItem:{
                ...this.state.deleteItem,
                id: '',
                fullName: ''
            }
        });
    };

    render() {
        return (
            <div>
                <form onKeyUp={this.handleKeyUp}>
                    <DataTable>
                        <DataTableHead>
                            <ContactsListHead
                                showCheckbox={this.props.showCheckboxList}
                                refreshContactsData={() => this.props.refreshContactsData()}
                            />
                            <ContactsListFilter
                                showCheckbox={this.props.showCheckboxList}
                                selectAllCheckboxes={() => this.props.selectAllCheckboxes()}
                                onSubmitFilter={this.props.onSubmitFilter}
                            />
                        </DataTableHead>
                        <DataTableBody>
                            {
                                this.props.contacts.length === 0 ? (
                                    <tr><td colSpan={11}>Geen contacten gevonden!</td></tr>
                                ) : (
                                    this.props.contacts.map((contact) => {
                                        return <ContactsListItem
                                            key={contact.id}
                                            {...contact}
                                            showCheckbox={this.props.showCheckboxList}
                                            checkedAllCheckboxes={this.props.checkedAllCheckboxes}
                                            showDeleteItemModal={this.showDeleteItemModal}
                                        />
                                    })
                                )
                            }
                        </DataTableBody>
                    </DataTable>
                </form>
                {
                    this.state.showDeleteItem &&
                        <ContactsDeleteItem
                            closeDeleteItemModal={this.closeDeleteItemModal}
                            {...this.state.deleteItem}
                        />
                }
            </div>
        );
    }
};

export default ContactsList;