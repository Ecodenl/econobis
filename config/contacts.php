<?php

return [

    'grid' => [

        /*
        |--------------------------------------------------------------------------
        | Address Type
        |--------------------------------------------------------------------------
        |
        | Geef hier het type adres op dat moet worden weergegeven in het grid
        |
        */

        'address_type_id' => 3,

    ],

];

//<?php
//
//return [
//
//    'grid' => 'test',
//
//];
