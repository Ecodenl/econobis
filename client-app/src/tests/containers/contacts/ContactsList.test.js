import React from 'react';
import { shallow } from 'enzyme';
import ContactsList from '../../../container/contact/list/ContactsList';
import contacts from '../../fixtures/contacts';

test('should render ContactsList with empty message', () => {
    const wrapper = shallow(<ContactsList contacts={[]} />);
    expect(wrapper).toMatchSnapshot();
});

test('should render ContactsList with contacts', () => {
    const wrapper = shallow(<ContactsList contacts={contacts} />);
    expect(wrapper).toMatchSnapshot();
});
