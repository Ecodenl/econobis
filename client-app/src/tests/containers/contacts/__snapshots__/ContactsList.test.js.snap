// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ContactsList with contacts 1`] = `
<div>
  <form
    onKeyUp={[Function]}
  >
    <DataTable>
      <DataTableHead>
        <Connect(ContactsListHead)
          refreshContactsData={[Function]}
        />
        <Connect(ContactsListFilter)
          selectAllCheckboxes={[Function]}
        />
      </DataTableHead>
      <DataTableBody>
        <Connect(ContactsListItem)
          city="Sint Pancras"
          createdAt={
            Object {
              "date": "2017-01-01",
            }
          }
          emailAddress="rob.rollenberg@xaris.nl"
          fullName="Rob Rollenberg"
          id={1}
          number="2017-1"
          phoneNumber="06-53987605"
          postalCode="1834 TK "
          showDeleteItemModal={[Function]}
          status="nieuw"
          streetAndNumber="Penant 34"
          typeName="Persoon"
        />
        <Connect(ContactsListItem)
          city="Alkmaar"
          createdAt={
            Object {
              "date": "2017-02-02",
            }
          }
          emailAddress="marco.tazelaar@xaris.nl"
          fullName="Marco Tazelaar"
          id={2}
          number="2017-2"
          phoneNumber="072-1234567"
          postalCode="1800 AA "
          showDeleteItemModal={[Function]}
          status="open"
          streetAndNumber="Straat 123"
          typeName="Persoon"
        />
        <Connect(ContactsListItem)
          city="Wognum"
          createdAt={
            Object {
              "date": "2017-03-03",
            }
          }
          emailAddress="info@xaris.nl"
          fullName="Xaris"
          id={2}
          number="2017-3"
          phoneNumber="088-0234567"
          postalCode="1687 AZ "
          showDeleteItemModal={[Function]}
          status="open"
          streetAndNumber="Westerspoor  1"
          typeName="Bedrijf"
        />
      </DataTableBody>
    </DataTable>
  </form>
</div>
`;

exports[`should render ContactsList with empty message 1`] = `
<div>
  <form
    onKeyUp={[Function]}
  >
    <DataTable>
      <DataTableHead>
        <Connect(ContactsListHead)
          refreshContactsData={[Function]}
        />
        <Connect(ContactsListFilter)
          selectAllCheckboxes={[Function]}
        />
      </DataTableHead>
      <DataTableBody>
        <tr>
          <td
            colSpan={11}
          >
            Geen contacten gevonden!
          </td>
        </tr>
      </DataTableBody>
    </DataTable>
  </form>
</div>
`;
